ecosystem/sep-0010.md

```diff
@@ -139,7 +139,80 @@ operation of the challenge is the **Home Domain**, and that the web auth domain
 GET <WEB_AUTH_ENDPOINT>
 ```
 
-Request Parameters:
+##### Request Header:
+
+Optionally, the server may forbid unauthorized calls to `GET <WEB_AUTH_ENDPOINT>`. This could be done to protect an
+endpoint against bad actors or limit access only to allowed applications.
+
+In that case, the client must add an `Authorization` header, and the server should respond to all requests with missing
+header with an error (see below). Server may choose to only accept calls from trusted applications, responding with an
+error to all other requests (see below).
+
+Authorized `WEB_AUTH_ENDPOINT` accepts `Authorization Bearer: <token>`, where token is [JSON Web Token](https://jwt.io/)
+authorizing the request.
+
+Client must specify the following claims:
+
+- `iat` (the time at which the JWT was issued
+  [RFC7519, Section 4.1.6](https://tools.ietf.org/html/rfc7519#section-4.1.6)) — current timestamp (for example
+  `1530644093`)
+- `exp` (the expiration time on or after which the JWT must not be accepted for processing,
+  [RFC7519, Section 4.1.4](https://tools.ietf.org/html/rfc7519#section-4.1.4))
+- `web_auth_endpoint` - should match the auth endpoint (`<WEB_AUTH_ENDPOINT>`)
+- All request's parameters (if specified in the request)
+
+Client must then correctly sign the payload with appropriate Stellar private key. To choose the private key client
+application should follow this steps:
+
+- If `client_domain` is specified, the token must be signed with the _Client Domain Account_ (i.e. [SEP-1] defined
+  `SIGNING_KEY`). (See [Verifying the Client Domain](#verifying-the-client-domain) section for more information)
+- Otherwise, the token must be signed with a private key of the **Client Account** passed in the `account` field.
+
+Token then should be signed with the selected private key, using **ed25519** algorithm. An appropriate header must be
+included
+
+```json
+{
+  "alg": "EdDSA"
+}
+```
+
+Example of the valid signed token in base64 format:
+
+```
+eyJhbGciOiJFZERTQSJ9.eyJpYXQiOjE3MTEwNzMyMjAsImV4cCI6MTcxMTA3NDEyMCwiYWNjb3VudCI6IkdDNlVDWFZUQU1ORzVKTE9NWkJTQ05ZWFZTTk5GSEwyM1NKUFlPT0ZKRTJBVllERFMyRkZUNDVDIiwiY2xpZW50X2RvbWFpbiI6ImV4YW1wbGUtd2FsbGV0LnN0ZWxsYXIub3JnIiwiaG9zdCI6ImV4YW1wbGUuY29tIiwicGF0aCI6Ii9zZXAxMC9hdXRoIn0.SJwDQBr_xjvb-Nx23t8xhnxzVZVY_IKCzTtPk0-xHfkse7sIyJK_Aobzq3JYpxZ5F5CpNEfOi4Yu5U4mDWVnDA
```

Comment:
it appears this token is missing the `web_auth_endpoint` claim

Reply:
it appears to be split in `host` and `path`, instead of a single `web_auth_endpoint` claim

Reply:
I'll update it, examples are from previous integration